ouroboros-consensus-byron/test/Test/ThreadNet/RealPBFT/TrackUpdates.hs

```diff
@@ -50,46 +54,52 @@ import qualified Ouroboros.Consensus.Byron.Ledger as Byron
 import qualified Test.ThreadNet.Ref.PBFT as Ref
 import           Test.ThreadNet.Network (TestNodeInitialization (..))
 import           Test.ThreadNet.Util.NodeJoinPlan
+import           Test.ThreadNet.Util.NodeTopology
 
 import           Test.ThreadNet.RealPBFT.ProtocolInfo
 
+-- | The expectation and observation regarding whether the hard-fork proposal
+-- succesfully updated the protocol version
+--
 data ProtocolVersionUpdateLabel = ProtocolVersionUpdateLabel
-  { pvuRequired :: !(Maybe Bool)
+  { pvuObserved :: !Bool
+    -- ^ whether the proposed protocol version is adopted or not adopted by the
+    -- end of the test
+  , pvuRequired :: !(Maybe Bool)
```

Comment:
Just wondering now why there are bangs here? That bang on the `Maybe` is unlikely to help much.

Reply:
Thomas and I chatted. We decided to leave strictness annotations off of test code fields unless empirically warranted. Basic idea: it should be as obvious as possible that little-to-no-effort was made to avoid space-leaks here. Lazily (har) putting a bang there without working to make it deep strictness might just give others (including the future self) a false sense of security.